### ISSUE TITLE
Add support for module loaders and unincluded files

### ISSUE THREAD

**Issue by nicojs** [stryker-mutator/stryker#115]
If we would like to mutation test stryker in stryker (and thereby eating our own dog food) we need support for module loaders.

Stryker (like all nodejs applications) uses the npm module loader, with the `require` function. Test files are responsible for loading there own files. Right now, the `mutate` array only supports files that are also in the `files` array, which is generally not bad, because it prevents confusion and really puts you in control of the order of the inclusion of the files in the testrunner. However, by forcing the test runner to load the source files, we also get the side effects. For example the `IsolatedTestRunnerAdapterWorker` will listen to events on the process (`process.on('message', /*omitted for clarity*/)`). This means that we have memory leak, because the listener never gets cleared, resulting in this error: `[2016-07-09 18:57:26.357] [ERROR] IsolatedTestRunnerAdapter - (node) warning: possible EventEmitter memory leak detected. 11 message listeners added. Use emitter.setMaxListeners() to increase limit.`

An other use case for having files that should not be included in the test runner is when using `karma` for angular applications for example. There some people package files that should be retrieved by the application, in the case of angular this can be the template files of views. Karma provides a solution by supporting (besides the plain string with a pattern) an object inside the `files` array. For example:

```javascript
  {pattern: 'app/index.html', included: false, served: false}
```
For more info see the [karma documentation](http://karma-runner.github.io/1.0/config/files.html)

I think we can use the same idea. So you can add the files with `included: false`. We should than also add files that are in `mutate`, but not in `files`, to the test runner sandboxes, but with `included: false`.

@simondel  What do you think?

**Comment by nicojs** [comment]
As was pointed out to me: this wouldn't necessarily fix our problem with running stryker on stryker, because the unit test for `IsolatedTestRunnerAdapterWorker ` would include `IsolatedTestRunnerAdapterWorker` and we would still have the memory leak. But as we don't have a unit test for `IsolatedTestRunnerAdapterWorker` the file would never be included :bowtie: (we do have an integration test for it)

If would ever create the  `IsolatedTestRunnerAdapterWorkerSpec` unit test file, it would be responsible for mocking away the `process.on` *before* requiring `IsolatedTestRunnerAdapterWorker`.

**Comment by simondel** [comment]
I would also add support for special files such as templates with the format `{pattern: 'app/index.html', included: false}`. However, I would still force users to enter files in both `mutate` and `files` if they should be mutated to make a clear separation between the files the testrunner needs and which of those files should be acted upon by our framework.

**Comment by nicojs** [comment]
Agreed. I will implement this feature.

**Comment by nicojs** [comment]
Fixed with #126
